### PR TITLE
Allows extensions to fall back on using an optional name in keystone

### PR DESCRIPTION
**Pull request by zack-shoylev**

### CODE REVIEW

**Comment by zack-shoylev** [comment]
Needed for https://issues.apache.org/jira/browse/JCLOUDS-1013?filter=-1

File: apis/openstack-keystone/src/test/java/org/jclouds/openstack/v2_0/functions/PresentWhenExtensionAnnotationNamespaceEqualsAnyNamespaceInExtensionsSetTest.java [hunk 1]
```diff
@@ -125,6 +138,23 @@ public void testPresentWhenAliasForExtensionMapsToNamespace() throws SecurityExc
 
    }
 
+   /**
+    * In devstack and going forward, namespaces are being deprecated. When namespace is missing (or replaced with a
+    * "fake" /fake namespace), allow matching by name and alias.
+    *
+    */
+   public void testPresentWhenNameSpaceIsMissingAndMatchByNameOrAlias() throws SecurityException, NoSuchMethodException {
+//      fail("its a no go");
```
Comment (nacx) [inline apis/openstack-keystone/src/test/java/org/jclouds/openstack/v2_0/functions/PresentWhenExtensionAnnotationNamespaceEqualsAnyNamespaceInExtensionsSetTest.java]: Remove this comment?

File: apis/openstack-keystone/src/test/java/org/jclouds/openstack/v2_0/functions/PresentWhenExtensionAnnotationNamespaceEqualsAnyNamespaceInExtensionsSetTest.java [hunk 2]
```diff
@@ -125,6 +138,23 @@ public void testPresentWhenAliasForExtensionMapsToNamespace() throws SecurityExc
...
+      Extension floatingIpsWithFakeNamespace = floatingIps.toBuilder()
+            .namespace(URI.create("http://docs.openstack.org/ext/fake"))
+            /*.name("Floating_ips")*/.build();
```
Comment (nacx) [inline apis/openstack-keystone/src/test/java/org/jclouds/openstack/v2_0/functions/PresentWhenExtensionAnnotationNamespaceEqualsAnyNamespaceInExtensionsSetTest.java]: Remove this one too?

**Comment by nacx** [comment]
Thanks @zack-shoylev! LGTM!

**Comment by nacx** [comment]
OH, is there any name that should be added by default to the existing apis?

**Comment by zack-shoylev** [comment]
@nacx We need at least FloatingIPApi because there is a jira for it, but there might be a few other extensions that we might want. I am thinking of adding names for the nova extensions and probably some of the neutron ones, but wanted to do it in a separate pr.

I will clean this up and merge, then double check which names are needed for the next PR.

**Comment by nacx** [comment]
+1 thanks!

**Comment by zack-shoylev** [comment]
merged

**Comment by zack-shoylev** [comment]
https://github.com/jclouds/jclouds/pull/892

**Pull request closed without merging**